homeassistant/components/metoffice/sensor.py

```diff
@@ -155,6 +155,8 @@ async def async_setup_entry(
     """Set up the Met Office weather sensor platform."""
     hass_data = hass.data[DOMAIN][entry.entry_id]
 
+    async_migrate_unique_ids(hass, hass_data)
```

Comment:
Please use the `entity_registry.async_migrate_entries()` helper instead.

Reply:
The issue I have with going via `entity_registry.async_migrate_entries()` is that process becomes essentially O(N^2) - i.e., for each entry you have to iterate through all possible names and check if it's the one.
With my current approach there is a single iteration through the list of names with a map lookup for each.

I know this is probably not a major issue but unless you feel strongly about it I would prefer to keep the current solution.

Reply:
I think it's more important to standardize.

Reply:
Okay, I found a way to do this in O(N) :)